wsagent/che-core-api-project/src/main/java/org/eclipse/che/api/search/server/impl/LuceneSearcher.java

```diff
@@ -437,9 +453,19 @@ public final void update(Path fsPath) throws ServerException {
   }
 
   private void doUpdate(Term deleteTerm, Path fsPath) throws ServerException {
+    if (!fsPath.toFile().exists()) {
+      return;
```

Comment:
can you make one ```if``` instead of two?

Reply:
It is done on purpose because those conditions are not logically related.